### PR TITLE
Add a way to directly modify the html code in Advanced mode

### LINKED ISSUE #10572: Code Mode is changing my HTML. It's not a true code mode.
### Mautic Version

4.0.0

### PHP version

7.4.25

### What browsers are you seeing the problem on?

Chrome

### What happened?

When I paste code into the GrapeJS editor with code mode as the theme, it makes a bunch of changes to the code to power the GrapeJS editor. ie. adding id's, auto-formatting. stripping out stuff. 

Code Mode is for people that want nothing to do with the editor Mautic provides because they generate their HTML from another source. ie: Litmus Email Builder, Custom implementation, etc ..

In my scenario, I have a custom build email builder that is completely unique to our business. It connects directly with our API so we can pull data and generate emails fast. (We are a music publishing company and often send emails containing lists of songs. Our builder allows us to just type in an ID and it generates an email with all the songs. Wayyyy out of scope for a Mautic template, but it works awesome for us.) Our builder spits out nice, formatted, minified HTML.

We just upgraded to Mautic 4 and now the code mode is all different and it's messing up the code. Adding ID's everywhere, all the minified formatting is lost.

I paste in this:
CleanShot 2021-11-03 at 09 42 48@2x

After save it looks like this:
CleanShot 2021-11-03 at 09 44 12@2x

A Code Mode option, and having the ability to edit HTML code for an email are not the same thing.

Am I missing something? 
CleanShot 2021-11-03 at 09 51 05@2x

### How can we reproduce this issue?

Step 1: Use Code Mode to past in email HTML.
Step 2: Save the email
Step 3: Open back up to review the code.
Step 4: Copy the code and string compared to what you pasted in. If you didn't implicitly press any button to format the code, it should be exactly what you pasted in.

### Relevant log output

_No response_

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

### LINKED ISSUE #10769: Code Mode isn't useable in GrapesJS
### Mautic Version

4.0.0

### PHP version

7.4

### What browsers are you seeing the problem on?

Not relevant

### What happened?

When building an email using the new GrapesJS builder, the user is given the option of selecting "Code Mode". In Code Mode, the user can paste in html code into the GrapesJS builder instead of MJML and the ui renders the email, seemingly as pasted into the box.

However, as [documented in this issue](https://github.com/mautic/mautic/issues/10475), GrapesJS actually rewrites the HTML, assigning IDs and classes and moving css into a style tag before the `</body>` tag. 

There is no way for a user to actually get the code they put in to the code editor come out the emails sent by Mautic, and it's nearly impossible to get an HTML email to render correctly using code mode in all email clients for this reason. 

Here is some example HTML that really brings home the issue:
```
<!doctype html>
<html xmlns="http://www.w3.org/1999/xhtml" xmlns:v="urn:schemas-microsoft-com:vml" xmlns:o="urn:schemas-microsoft-com:office:office">

<head>
    <title>
        {subject}
    </title>
</head>
<body style="background-color:#ffffff;">
  <h2 style="color:red;font-size:18px;font-family:serif;">This is a red serif heading</h2>
  <p><a href="https://www.mautic.org" style="color:white; text-decoration:none; padding:8px 20px; background:orange;">This is an orange button</a></p>
  </body>
</html>
```

Here is how that renders with the old code mode (non-grapes-js) in gmail, essentially just passing it through:
image

Here is the code that the new grapes-js editor saves (reformatted for readability):
```
<!DOCTYPE html>
<head>
   <title>
      Demo Code Mode
   </title>
</head>
<body>
   <h2 id="ikph">This is a red serif heading</h2>
   <p><a href="https://www.mautic.org" id="iysd">This is an orange button</a></p>
   <style>#ikph{color:red;font-size:18px;font-family:serif;}#iysd{color:white;text-decoration:none;padding:8px 20px;background:orange;}</style>
</body>
```

Here is how that same code renders in gmail with the new code mode (grapes-js):
image

I interpret this as a bug/regression for either or both of the following reasons:
- Users can't actually use "code mode" to make an email with style that renders correctly in gmail, so we have a feature to make an HTML email that can't actually make an HTML email
- The expectation of users is that the new code mode has at least the same level of functionality as the old code mode - I can paste code into it and get the same code back out when an email is sent.

The way that issue #10475 was closed, it's implied that this is a feature  of GrapesJS, not a bug in Mautic, but I think interpreting it that way is a disservice to our users. I would think that any of the following three would be a better resolution, especially since the [GrapesJS project has implied this is the intended behavior of their editor](https://github.com/artf/grapesjs/issues/546):
- If GrapesJS is enabled, disable Code Mode until a more usable experience can be developed
- If GrapesJS is enabled, make code mode simple textarea in which a user can paste in HTML code that will be sent, without modification, as an email. No preview, no fancy code editor, just a textarea. The resulting email can be previewed using the preview button after saving. 
- After enabling GrapesJS, make Code Mode use the old editor, while templates use the new GrapesJS editor, at least until a new option can be developed
- Use GrapesJS for templates, but for code mode, use another tool or create a new simple preview from scratch that doesn't modify the code pasted in

### How can we reproduce this issue?

Step 1: Enable GrapesJS
Step 2: Create a new email and select "Code Mode"
Step 3: Click the builder and then the "Edit Code" button
Step 4: Paste in the following HTML
```
<!doctype html>
<html xmlns="http://www.w3.org/1999/xhtml" xmlns:v="urn:schemas-microsoft-com:vml" xmlns:o="urn:schemas-microsoft-com:office:office">

<head>
    <title>
        {subject}
    </title>
</head>
<body style="background-color:#ffffff;">
  <h2 style="color:red;font-size:18px;font-family:serif;">This is a red serif heading</h2>
  <p><a href="https://www.mautic.org" style="color:white; text-decoration:none; padding:8px 20px; background:orange;">This is an orange button</a></p>
  </body>
</html>
```
Step 5: Save and close the email
Step 6: Use the preview URL or send yourself a test and compare the code you get out to the code you put in

### Relevant log output

_No response_

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by volha-pivavarchyk**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [no]
| New feature/enhancement? (use the a.x branch)      | [yes]
| Deprecations?                          | [no]
| BC breaks? (use the c.x branch)        | [no]
| Automated tests included?              | [no] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes [#10769](https://github.com/mautic/mautic/issues/10769) 

#### Description:
If GrapesJS is enabled, a simple textarea (emailform_customHtml)  field is displayed in the advanced tab that can be used to to directly add HTML. The textarea is only visible if code mode was selected. The builder button to launch GrapesJS is disabled.

image

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a new email
3. Select the code mode email type
4. The builder button is disabled
5. Select the advanced tab
6. The HTML Code field is displayed

<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10976"></a>

### CODE REVIEW

File: plugins/GrapesJsBuilderBundle/Assets/library/js/builder.js [hunk 1]
```diff
@@ -70,11 +70,42 @@ function setThemeHtml(theme) {
   });
 }
 
+/**
+ * The builder button to launch GrapesJS will be disabled when the code mode theme is selected
+ *
+ * @param theme
+ */
+function switchBuilderButton(theme) {
```
Comment (kuzmany) [inline plugins/GrapesJsBuilderBundle/Assets/library/js/builder.js:78]: I don't know If GrapesJsBuilderBundle is right place to put it.
We need cover it for emails and pages in general not related what builder is enabled.
I understand legacy builder is removed in M5 but If we push this now, we need rewrite it in M5 again.

**Review by kuzmany — APPROVED** [review]
Even If I am not okay with the change, we need fixed it soon as possible. 
Then let's go to merge it and improve it later.

Works for me.

**Review by kuzmany — CHANGES_REQUESTED** [review]
Does not work for pages (no Advanced tab there)

**Comment by volha-pivavarchyk** [comment]
Will it be good solution if I add the Advanced tab?

**Comment by volha-pivavarchyk** [comment]
I added the advanced tab in the edit page

**Comment by kuzmany** [comment]
Works. Probably for future we can add HTML to separate tab Code Mode for both email/pages to make it more clear

**Review by kuzmany — DISMISSED** [review]
Works. 
Probably we can add HTML to separate tab Code Mode for both email/pages to make it more clear. But it's up to community. I am gonna approve it for now

**Review by escopecz — DISMISSED** [review]
I see no issue with the code 👍

**Comment by ghost** [comment]
I also tested this issue in Gitpod and find one problem. When I choose code mode Builder is still active. So I opened this and seen GrapeJS Builder. After closed builder I can't select any theme. It looks like JS bug.

@volha-pivavarchyk looks Builder is still active. I have chosen code mode.
image

**Comment by kuzmany** [comment]
@biggala2310 seems like button is not disabled for mobile version of view.  Works on desktop properly
@volha-pivavarchyk can you check it?

**Comment by ghost** [comment]
@kuzmany This is only first bug. Second is to change themes.
Step to reproduce:
1. Choose HTML mode
2. Choose Blank mode
3. Go to Builder and exit
4. Try change theme - it is impossible.

**Comment by volha-pivavarchyk** [comment]
@biggala2310  @kuzmany I checked and fixed the issue for mobile version. 

@biggala2310 I couldn't reproduce the second bug it in my local version, only in Gitpod. And I've checked this bug is reproduced in another PR in Gitpod as well.

**Review by kuzmany — DISMISSED** [review]
Works on mobile too.

**Comment by renatocron** [comment]
I just tested this locally, I was able to change the theme after using Code editor, both when saving and closing, and when just changing without saving

**Comment by escopecz** [comment]
@volha-pivavarchyk I changed the target branch from 4.x to 4.2 so we could release this in 4.2.2. Do you think you could rebase and fix the conflict? Or if you wish to release in 4.3 then we can switch back to 4.x and merge right away.

**Comment by volha-pivavarchyk** [comment]
@escopecz I rebased and didn't get a good result. Maybe the best way is to reopen the PR to branch 4.2 or 4.x?

**Comment by volha-pivavarchyk** [comment]
Rebase gone wrong

**Pull request closed without merging**